stdlib/SuiteSparse/src/cholmod.jl

```diff
@@ -798,13 +798,25 @@ get_perm(FC::FactorComponent) = get_perm(Factor(FC))
 # Conversion/construction
 function Dense{T}(A::StridedVecOrMat) where T<:VTypes
     d = allocate_dense(size(A, 1), size(A, 2), stride(A, 2), T)
-    s = unsafe_load(pointer(d))
-    for i in eachindex(A)
-        unsafe_store!(s.x, A[i], i)
+    GC.@preserve d begin
+        s = unsafe_load(pointer(d))
+        for i in eachindex(A)
+            unsafe_store!(s.x, A[i], i)
+        end
     end
     d
 end
-function Dense(A::StridedVecOrMat)
+function Dense{T}(A::Union{Adjoint{<:Any, <:StridedVecOrMat}, Transpose{<:Any, <:StridedVecOrMat}}) where T<:VTypes
+    d = allocate_dense(size(A, 1), size(A, 2), size(A, 1), T)
+    GC.@preserve d begin
+        s = unsafe_load(pointer(d))
+        for (i, c) in enumerate(eachindex(A))
+            unsafe_store!(s.x, A[c], i)
```

Comment:
Ok, I misread that earlier. It works, because the `i`s run from 1 to `length(A)`, and not through the indices of `A` (like `c`). I wonder if we should take the same approach in other method above, I mean with `enumerate(eachindex(A))` etc.